api/article/views.py

```diff
@@ -7,8 +9,23 @@ class ArticleViewSet(viewsets.ModelViewSet):
     serializer_class = ArticleSerializer
     permission_classes = permissions.IsAuthenticatedOrReadOnly,
 
+    serializer_class_map = {
+        'retrieve': ArticleSerializer,
+        'list': ArticleSerializer,
+        'create': ArticleSerializer,
+        'self_articles': ArticleSerializer,
+        'partial_update': ArticleSerializer,
+    }
+
+    def get_serializer_class(self):
+        return self.serializer_class_map[self.action]
+
     def get_queryset(self):
-        return Article.objects.prefetch_related('author').order_by('-created')
+        author = self.request.user
+        articles = Article.objects.prefetch_related('author').order_by('-created')
```

Comment:
Perhaps better way is to move `order_by` as default ordering to `Article`. What do you think?

Reply:
Yes, that's right, it was a temporary solution.